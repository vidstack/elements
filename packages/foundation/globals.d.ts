/// <reference types="./dom" />

// -------------------------------------------------------------------------------------------
// Global Events
// -------------------------------------------------------------------------------------------

import type { ContextConsumerConnectEvent } from './types/context';
import type { FullscreenEvents } from './types/fullscreen';
import type { LoggerEvents } from './types/logger/events';
import type { ScreenOrientationEvents } from './types/screen-orientation';

declare global {
  interface GlobalEventHandlersEventMap
    extends FullscreenEvents,
      ScreenOrientationEvents,
      LoggerEvents {
    'vds-noop': any;
    'vds-context-consumer-connect': ContextConsumerConnectEvent;
  }
}
