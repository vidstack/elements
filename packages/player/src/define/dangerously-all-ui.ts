import './vds-aspect-ratio';
import './vds-gesture';
import './vds-poster';
// Buttons
import './vds-toggle-button';
import './vds-mute-button';
import './vds-play-button';
import './vds-fullscreen-button';
// Sliders
import './vds-slider';
import './vds-time-slider';
import './vds-volume-slider';
import './vds-slider-video';
import './vds-slider-value-text';
// Time
import './vds-time';
