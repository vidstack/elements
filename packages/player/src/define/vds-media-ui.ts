import { safelyDefineCustomElement } from '@vidstack/foundation';

import { MediaUiElement } from '../media/ui/MediaUiElement';

safelyDefineCustomElement('vds-media-ui', MediaUiElement);

declare global {
  interface HTMLElementTagNameMap {
    'vds-media-ui': MediaUiElement;
  }
}

if (__DEV__) {
  // TODO: add release notes link.
  console.warn('`<vds-media-ui>` has been deprecated and will be removed in 1.0.');
}
