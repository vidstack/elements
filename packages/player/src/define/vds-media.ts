import { safelyDefineCustomElement } from '@vidstack/foundation';

import { MediaElement } from '../media/MediaElement';

safelyDefineCustomElement('vds-media', MediaElement);

declare global {
  interface HTMLElementTagNameMap {
    'vds-media': MediaElement;
  }
}
