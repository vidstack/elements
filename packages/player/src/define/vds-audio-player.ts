import './vds-media-ui';

import { safelyDefineCustomElement } from '@vidstack/foundation';

import { AudioPlayerElement } from '../players/audio/AudioPlayerElement';

safelyDefineCustomElement('vds-audio-player', AudioPlayerElement);

declare global {
  interface HTMLElementTagNameMap {
    'vds-audio-player': AudioPlayerElement;
  }
}

if (__DEV__) {
  // TODO: add release notes link.
  console.warn('`<vds-audio-player>` has been deprecated and will be removed in 1.0.');
}
