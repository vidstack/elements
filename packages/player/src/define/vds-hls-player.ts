import './vds-media-ui';

import { safelyDefineCustomElement } from '@vidstack/foundation';

import { HlsPlayerElement } from '../players/hls/HlsPlayerElement';

safelyDefineCustomElement('vds-hls-player', HlsPlayerElement);

declare global {
  interface HTMLElementTagNameMap {
    'vds-hls-player': HlsPlayerElement;
  }
}

if (__DEV__) {
  // TODO: add release notes link.
  console.warn('`<vds-hls-player>` has been deprecated and will be removed in 1.0.');
}
