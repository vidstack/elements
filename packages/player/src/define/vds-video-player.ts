import './vds-media-ui';

import { safelyDefineCustomElement } from '@vidstack/foundation';

import { VideoPlayerElement } from '../players/video/VideoPlayerElement';

safelyDefineCustomElement('vds-video-player', VideoPlayerElement);

declare global {
  interface HTMLElementTagNameMap {
    'vds-video-player': VideoPlayerElement;
  }
}

if (__DEV__) {
  // TODO: add release notes link.
  console.warn('`<vds-video-player>` has been deprecated and will be removed in 1.0.');
}
