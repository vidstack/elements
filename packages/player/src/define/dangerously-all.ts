import './vds-media';
import './vds-media-sync';
import './vds-media-visibility';
import './vds-audio';
import './vds-hls';
import './vds-video';
import './dangerously-all-ui';
