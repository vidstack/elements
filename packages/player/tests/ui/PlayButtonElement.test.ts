import '$lib/define/vds-play-button';

import { elementUpdated } from '@open-wc/testing-helpers';
import { waitForEvent } from '@vidstack/foundation';
import { html } from 'lit';

import { buildMediaPlayerFixture } from '$test-utils';

async function buildFixture() {
  const { media, provider } = await buildMediaPlayerFixture(html`
    <vds-play-button>
      <div class="play"></div>
      <div class="pause"></div>
    </vds-play-button>
  `);

  const button = media.querySelector('vds-play-button')!;

  return { media, provider, button };
}

it('should render light DOM', async function () {
  const { button } = await buildFixture();
  expect(button).dom.toMatchSnapshot();
});

it('should render shadow DOM', async function () {
  const { button } = await buildFixture();
  expect(button).shadowDom.to.equal(`<slot></slot>`);
});

it('should update paused state', async function () {
  const { media, button } = await buildFixture();

  media.controller._store.paused.set(true);
  await elementUpdated(button);

  expect(button.pressed).to.be.false;
  expect(button.getAttribute('aria-pressed')).to.equal('false');
  expect(button.hasAttribute('media-paused')).to.be.true;

  media.controller._store.paused.set(false);
  await elementUpdated(button);

  expect(button.pressed).to.be.true;
  expect(button.getAttribute('aria-pressed')).to.equal('true');
  expect(button.hasAttribute('media-paused')).to.be.false;
});

it('should play', async function () {
  const { media, provider, button } = await buildFixture();

  const pausedSpy = vi.spyOn(provider, '_paused', 'set');

  media.controller._store.paused.set(true);
  await elementUpdated(button);

  setTimeout(() => button.dispatchEvent(new MouseEvent('pointerdown')));

  await waitForEvent(button, 'vds-play-request');
  expect(pausedSpy).toHaveBeenCalledWith(false);
});

it('should pause', async function () {
  const { media, provider, button } = await buildFixture();

  const pausedSpy = vi.spyOn(provider, '_paused', 'set');

  media.controller._store.paused.set(false);
  await elementUpdated(button);

  setTimeout(() => button.dispatchEvent(new MouseEvent('pointerdown')));

  await waitForEvent(button, 'vds-pause-request');
  expect(pausedSpy).toHaveBeenCalledWith(true);
});
