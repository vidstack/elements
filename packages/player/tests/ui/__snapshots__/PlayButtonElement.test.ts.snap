// Vitest Snapshot v1

exports[`should render light DOM 1`] = `
<vds-play-button
  aria-label="Play"
  aria-pressed="false"
  paused=""
  role="button"
  tabindex="0"
>
  
      
  <div
    class="play"
  />
  
      
  <div
    class="pause"
  />
  
    
</vds-play-button>
`;
